src/memory_map.rs

```diff
@@ -23,9 +23,15 @@ pub struct MemoryMapTag {
 impl MemoryMapTag {
     /// Return an iterator over all AVAILABLE marked memory areas.
     pub fn memory_areas(&self) -> MemoryAreaIter {
```

Comment:
I wonder if, rather than having two iteratory types, this could return `impl Iterator<Item = &MemoryArea` (maybe you'll need a lifetime idk) and then the body could be `self.all_memory_areas().filter(|entry| entry.typ == 1)`? I think that'd be cleaner, if it works.

Reply:
That would be cleaner and it does work. Initially I was worried that filter wouldn't work without the standard library (I'm using this crate for work on an OS kernel where I don't have the standard library and I'm new to working without the standard library, so I haven't yet figured out what is part of the standard library vs what is part of core), so I chose to avoid. On further investigation, it seems to work fine, so I will update my pull request to make use of fileter.